bin/knc.c

```diff
@@ -999,40 +1007,41 @@ write_network_buffer(work_t *work)
 	return len;
 }
 
-#define MAX(a,b)	(((a) > (b)) ? (a) : (b))
-
 int
 move_data(work_t *work)
 {
+	struct pollfd pfds[4];
+	nfds_t nfds;
+	struct pollfd * const pfd_net = pfds+0;
+	struct pollfd * const pfd_in  = pfds+1;
+	struct pollfd * const pfd_out = pfds+2;
+	struct pollfd * const pfd_err = pfds+3;
 	int		ret;
 	int		mret;
-	int		select_is_the_worst_api_ever;
+	int		timeout;
 	fd_set		rdset;
 	fd_set		wrset;
 	char		local_active = 1;
 	char		network_active = 1;
 	char		shut_nread_lwrite = 0;
 	char		shut_nwrite_lread = 0;
-	struct timeval	tv;
+	char		pollrdhup_net = 0;
+	char		pollrdhup_in = 0;
 	char		errbuf[8192];
 
+	memset(pfds, 0, sizeof(pfds));
+	pfd_net->fd = work->network_fd;
+	pfd_in->fd  = work->local_in;
+	pfd_out->fd = work->local_out;
+	pfd_err->fd = work->local_err;
+	nfds = (-1 == work->local_err) ? 3 : 4;
```

Comment:
Clearly it's not Roland's style to put the literal on the left of the comparison operator.

Since modern C compilers warn about assignments in conditionals, putting the literal on the left is not that useful for avoiding accidental assignment typos.

Reply:
moved literal to right-hand-side in two places